src/UiPatternsManager.php

```diff
@@ -138,6 +140,24 @@ protected function alterDefinitions(&$definitions) {
   /**
    * {@inheritdoc}
    */
+  public function getDefinitions() {
+    $definitions = $this->getCachedDefinitions();
+    if (!isset($definitions)) {
+      // Remove derivative id from pattern definitions keys.
```

Comment:
Yeah this is a controversy point! :D I'm also not a super fan of this but all the system has been built on the assumption that pattern IDs are unique in the system and this should be kept regardless of the way we fetch patterns definition since it allows for a nice dev UX. It would also be quite disruptive in terms of backwards compatibility (patterns IDs are stored into configuration, used as template file names, etc.).

That said I'm open for suggestions!